File: day-5/day-5-b.py
# -*- coding: utf-8 -*-

tot=0

rs=[]

read_range=True

with open('input.txt') as f:

	for line in f:

		if line=='\n':

			read_range=False

		else:

			if read_range==True:

				l = list(map(int, line.rstrip().split('-')))

				if len(rs)==0:

					rs.append(l)

					print(rs)

				else:

					print(l)

					i=0

					for r in rs:

						#print(str(i) + " : " + str(r))

						i=i+1



						if l[1] < r[0] or l[0] > r[1]:
							rs.append(r)

						elif not (l[0] >= r[0] and l[1] <= r[1]):

							if l[0] < r[0] and l[1] > r[0]:
								rs[i][0]=l[0]

							if l[0] < r[1] and l[1] > r[1]:
								rs[i][1]=l[1]

	print(rs)

File: day-5/day-5-b.js
const fs = require('fs')

fs.readFile('input.txt', 'utf8', (err, data) => {

  const lines = data.split('\n')

  const ranges = lines.filter(l => l.includes('-')).map(l => l.split('-'))

  const new_ranges = ranges.reduce((acc, r) => {



    if (acc.length==0) {

      acc = [r]

    } else {

      let overlap = false;

      //console.log('==========')

      //console.log(r)
      //console.log(acc)

      acc.forEach((a, i) => {

        if (r[1] < a[0] || r[0] > a[1]) {

          //acc = [...acc, r]

          //console.log("A : " + a)



        } else if (r[0] >= a[0] && r[1] <= a[1]) {

          //console.log("B : " + a)

          overlap=true



        } else {


          if (r[0] < a[0] && r[1] > a[0]) {
            acc[i][0]=r[0]
            //out=true

            //console.log("C : " + a)

            overlap=true
          }

          if (r[0] < a[1] && r[1] > a[1]) {
            acc[i][1]=r[1]
            //out=true

            //console.log("D : " + a)

            overlap=true

          }

        }



      })

      if (!overlap) {

        acc = [...acc, r]
      }

      /*if (!out) {
        acc = [...acc, r]
      }*/






    }

    return acc

  }, [])

  console.log(new_ranges)

  console.log(new_ranges.reduce((acc, r) => {


    acc = acc + (Number(r[1])-Number(r[0])) + 1


    return acc;



  }, 0))







})
